components/global/Header/HeaderAccount.tsx

```diff
@@ -1,14 +1,96 @@
 import React from "react";
 import SignIn from "components/scaffolding/Auth/SignIn";
 import SignOut from "components/scaffolding/Auth/SignOut";
+import * as Styled from "./Header.styles";
+import { Dropdown, NamedLink } from "components/atomic";
+import { Authorize } from "components/auth";
+import { useTranslation } from "react-i18next";
+import { RouteHelper } from "routes";
+import useIsAuthenticated, { useSignInOut } from "hooks/useIsAuthenticated";
+type NamedLinkProps = React.ComponentProps<typeof NamedLink>;
+type AuthorizeProps = React.ComponentProps<typeof Authorize>;
+
+interface HeaderNavParent extends HeaderNavItem {
+  route?: never;
+  children: HeaderNavLink[];
+}
+
+interface HeaderNavLink extends HeaderNavItem {
+  route: NamedLinkProps["route"];
+  children?: never;
+}
+
+interface HeaderNavItem {
+  label: string;
+  actions?: AuthorizeProps["actions"];
+}
 
 const HeaderAccount = () => {
-  return (
-    <>
-      <SignIn />
```

Comment:
These components struck me as unnecessary, and I've been trying to figure out how to articulate why. I think it was in part because I learned React early in my dev journey, so it seems much more natural to conditionally render the label on a button and its onClick than to have two separate buttons that use css to hide/show. It seemed straightforward to combine the logic of both of these components into one hook and not use the components. BUT when I implemented it, it essentially converged on your idea of rolling them into a single component.